rust/src/lib/nm/settings/ip.rs

```diff
@@ -27,6 +27,19 @@ fn gen_nm_ipv4_setting(
                 let mut nm_setting = NmSettingIp::default();
                 nm_setting.method = Some(NmSettingIpMethod::Disabled);
                 nm_conn.ipv4 = Some(nm_setting);
+            } else if let Some(setting) = nm_conn.connection.as_ref() {
+                // if the network state only contains local route rule absent,
+                // we need to specify the ip setting for
+                // loopback with replace_local_rule set a false
+                if let Some(iface_name) = setting.iface_name.clone() {
+                    if iface_name.eq(&"lo") {
+                        let mut nm_setting = NmSettingIp::default();
+                        nm_setting.method = Some(NmSettingIpMethod::Manual);
+                        nm_setting.addresses = vec!["127.0.0.1/8".to_string()];
+                        nm_setting.replace_local_rule = Some(true);
+                        nm_conn.ipv4 = Some(nm_setting);
+                    }
+                }
```

Comment:
There can be a situation where the user only specified the following:

```
route-rules:
  config:
  - priority: 0
    route-table: 255
    state: absent
```

Here `iface_ip` will be `None` and thefore it would skip the generation of ipv4_setting which is wrong. This why the extra code is needed.

Reply:
Then you should fix `append_route_rule()` and `apply_absent_rule()` instead of doing similar work in other place.